homeassistant/components/zamg/weather.py

```diff
@@ -52,93 +40,114 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the ZAMG weather platform."""
-    name = config.get(CONF_NAME)
+    LOGGER.warning(
+        "Configuration of the zamg integration in YAML is deprecated and "
+        "will be removed in a further release of Home Assistant; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+
+    probe = ZamgData()
+
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    station_id = config.get(CONF_STATION_ID) or probe.closest_station(
+        latitude, longitude
+    )
+    probe.set_default_station(station_id)
 
-    station_id = config.get(CONF_STATION_ID) or closest_station(
-        latitude, longitude, hass.config.config_dir
+    async_add_entities(
+        [ZamgWeather(probe, probe.get_data("Name", station_id), station_id)]
```

Comment:
I think `async_setup_platform` should just call the import method of your config flow (I didn't see it being called anywhere else) and not create entities anymore.
Same for the sensor platform.

Reply:
done ;-)